src/components/builds/BuildCard.tsx

```diff
@@ -0,0 +1,202 @@
+import BuildBranchNameChipNew from '../chips/BuildBranchNameChipNew';
+import RepositoryNameChipNew from '../chips/RepositoryNameChipNew';
+import RepositoryOwnerChipNew from '../chips/RepositoryOwnerChipNew';
+import usePageWidth from '../../utils/usePageWidth';
+import { navigateBuildHelper } from '../../utils/navigateHelper';
+import { muiThemeOptions } from '../../cirrusTheme';
+
+import { BuildCard_build$key } from './__generated__/BuildCard_build.graphql';
+
+const useStyles = makeStyles(theme => {
+  return {
+    card: {
+      cursor: 'pointer',
+      '&:hover': {
+        backgroundColor: theme.palette.action.hover,
+      },
+    },
+    // For pages with chart
+    cardSelected: {
+      backgroundColor: theme.palette.action.hover,
+    },
+    commitName: {
+      overflow: 'hidden',
+      textOverflow: 'ellipsis',
+      display: '-webkit-box',
+      WebkitLineClamp: 1,
+      WebkitBoxOrient: 'vertical',
+      whiteSpace: 'normal',
+      [theme.breakpoints.only('xs')]: {
+        marginBottom: theme.spacing(2.5),
+      },
+    },
+  };
+});
+
+interface Props {
+  build: BuildCard_build$key;
+  // For pages with chart
+  selectedBuildId?: boolean;
+  setSelectedBuildId?: Function;
+}
+
+export default function BuildCard(props: Props) {
+  let build = useFragment(
+    graphql`
+      fragment BuildCard_build on Build {
+        id
+        changeMessageTitle
+        repository {
+          ...RepositoryNameChipNew_repository
+          ...RepositoryOwnerChipNew_repository
+        }
+        ...Hash_build
+        ...Duration_build
+        ...BuildStatusChipNew_build
+        ...BuildBranchNameChipNew_build
+      }
+    `,
+    props.build,
+  );
+
+  let classes = useStyles();
+  let theme = useTheme();
+  const navigate = useNavigate();
+
+  const pageWidth = usePageWidth();
+
+  let isMdScreenWidth = pageWidth >= theme.breakpoints.values.md;
+
+  const themeOptions = useRecoilValue(muiThemeOptions);
+  const muiTheme = useMemo(() => createTheme(themeOptions), [themeOptions]);
+
+  // For pages with chart
+  let rowProps;
+  const selectable = !!props.setSelectedBuildId;
+  if (selectable) {
+    rowProps = {
+      onMouseEnter() {
+        if (props.selectedBuildId) return;
+        props.setSelectedBuildId(build.id);
+      },
+      onMouseLeave() {
+        props.setSelectedBuildId(null);
+      },
+    };
+  }
+
+  return (
+    <ThemeProvider theme={muiTheme}>
+      <Grid
+        className={cx(classes.card, props.selectedBuildId && classes.cardSelected)}
+        container
+        columns={4}
+        direction={{ xs: 'column', sm: 'row' }}
+        spacing={{ xs: 0.5, sm: 1, md: 3 }}
+        mt={{ sm: 0.5, md: 1.5 }}
+        alignItems={{ xs: 'start', sm: 'center' }}
+        sx={{
+          borderBottom: '1px solid',
+          borderColor: 'divider',
+        }}
+        onClick={e => {
+          const target = e.target as HTMLElement;
+          if (target.closest('a')) return;
```

Comment:
Why is this needed?

Reply:
To click on the row of the build (with navigation to the build) doesn’t overlap a click on a chip (with navigation somewhere else).